### PR TITLE
fix: bugs in oplsNipals and OPLS + testcases

### PR DESCRIPTION
**Pull request by targos**
- fix: bugs in oplsNipals and OPLS + testcases
- chore: update dependencies
- feat: returned binary classifiers function
- chore: fix lint-prettier
- feat: add predictCategory method + testcase
- chore: suggestions implemented

### CODE REVIEW

File: src/OPLS.js [hunk 1]
```diff
@@ -358,57 +415,64 @@ export class OPLS {
...
       this.tssy = tss(labels);
       this.tssx = tss(features);
     }
-
     const oplsC = oplsNipals(features, labels);
     const plsC = new NIPALS(oplsC.filteredX, { Y: labels });
-
-    const tPred = oplsC.filteredX.mmul(plsC.w.transpose());
+    const tPred = plsC.t.clone();
+    // const yHat = tPred.mmul(plsC.betas).mmul(plsC.q.transpose()); // ok
```
Comment (lpatiny) [inline src/OPLS.js:475]: You should remove this comment if not used.

File: src/OPLS.js [hunk 2]
```diff
@@ -212,18 +211,28 @@ export class OPLS {
     const tOrthCV = this.tOrthCV;
     const yHatCV = this.yHatCV;
     const m = this.model[nc - 1];
-    const XOrth = m.XOrth;
+    const XOrth = new Matrix(features.rows, features.columns);
```
Comment (lpatiny) [inline src/OPLS.js:214]: any reason it starts with an uppercase ? Variables are expected to start with lowercase.

File: src/OPLS.js [hunk 3]
```diff
@@ -358,57 +415,64 @@ export class OPLS {
...
+      for (let i = 0; i < components.rows; i++) {
+        totalPred.setRow(i, [components.getRowVector(i).sum()]);
+      }
     }
 
     if (labels?.rows > 0) {
       if (this.mode === 'regression') {
         const tssy = tss(labels);
-        const press = tss(labels.clone().sub(yHat));
+        const press = tss(labels.clone().sub(totalPred));
```
Comment (lpatiny) [inline src/OPLS.js:430]: Would it be possible in general to use longer variable name to help to understand the code ?

**Review by lpatiny — COMMENTED** [review]
If I understand well the code was compared with a R implementation and it gives the same result. Is this correct @josoriom ?

If possible it would be nice to have some longer name for the variables so that their meanings are obvious.

Are you also able to fix the merge conflict ?

File: src/OPLS.js (hunk 3 above)
Comment (josoriom) [reply src/OPLS.js:430]: Would be nice! I don't know what @jwist think about it?

**Pull request merged**